{"url": "https://api.github.com/repos/datalad/datalad/issues/2663", "repository_url": "https://api.github.com/repos/datalad/datalad", "labels_url": "https://api.github.com/repos/datalad/datalad/issues/2663/labels{/name}", "comments_url": "https://api.github.com/repos/datalad/datalad/issues/2663/comments", "events_url": "https://api.github.com/repos/datalad/datalad/issues/2663/events", "html_url": "https://github.com/datalad/datalad/pull/2663", "id": 332875218, "node_id": "MDExOlB1bGxSZXF1ZXN0MTk1MjUyODEw", "number": 2663, "title": "BF: Handle 401 errors from LORIS tokens", "user": {"login": "driusan", "id": 498329, "node_id": "MDQ6VXNlcjQ5ODMyOQ==", "avatar_url": "https://avatars.githubusercontent.com/u/498329?v=4", "gravatar_id": "", "url": "https://api.github.com/users/driusan", "html_url": "https://github.com/driusan", "followers_url": "https://api.github.com/users/driusan/followers", "following_url": "https://api.github.com/users/driusan/following{/other_user}", "gists_url": "https://api.github.com/users/driusan/gists{/gist_id}", "starred_url": "https://api.github.com/users/driusan/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/driusan/subscriptions", "organizations_url": "https://api.github.com/users/driusan/orgs", "repos_url": "https://api.github.com/users/driusan/repos", "events_url": "https://api.github.com/users/driusan/events{/privacy}", "received_events_url": "https://api.github.com/users/driusan/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 5, "created_at": "2018-06-15T18:34:36Z", "updated_at": "2018-06-28T15:46:26Z", "closed_at": "2018-06-28T15:46:26Z", "author_association": "COLLABORATOR", "active_lock_reason": null, "pull_request": {"url": "https://api.github.com/repos/datalad/datalad/pulls/2663", "html_url": "https://github.com/datalad/datalad/pull/2663", "diff_url": "https://github.com/datalad/datalad/pull/2663.diff", "patch_url": "https://github.com/datalad/datalad/pull/2663.patch"}, "body": "When a request is made with an incorrect username/password,\r\nurllib.urlopen raises an HTTPError exception. Since datalad\r\nis looking for AccessDeniedError exceptions in order to decide\r\nwhether to re-prompt for new credentials, it doesn't get caught\r\nand there's no easy way to enter the correct credentials if\r\nyou don't get it right the first time.\r\n\r\nThis converts the implicit exception in LORIS token generator\r\nto an AccessDeniedError.", "closed_by": {"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/datalad/datalad/issues/2663/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/datalad/datalad/issues/2663/timeline", "performed_via_github_app": null}